pkg/operator/csi/csistorageclasscontroller/csi_storageclass_controller.go

```diff
@@ -83,6 +94,7 @@ func NewCSIStorageClassController(
 	).WithInformers(
 		operatorClient.Informer(),
 		informerFactory.Storage().V1().StorageClasses().Informer(),
+		operatorInformer.Operator().V1().ClusterCSIDrivers().Informer(),
```

Comment:
This is basically the same informer as `operatorClient.Informer()`, could this be the reason of the problem you're facing?

Reply:
Discussed offline, I think we need this for the reasons Hemant pointed out, and my bug was elsewhere.